### DISCUSSION TITLE
おすすめTL

### DISCUSSION THREAD

**Discussion by penginn-net** [yojo-art/cherrypick#370] [General]
#12 
何を持っておすすめと判断するのか

**Comment by penginn-net** [comment]
https://github.com/yojo-art/cherrypick/issues/12#issuecomment-2294944046
>自分がよく使うリアクション、フォローユーザーがよく使うリアクションあたりから抽出して自動設定、手動でお気に入りを固定する機能があると良さそう
全部統計すると重そうだから一定期間内にサンプル数が規定量を超えるようなら適当な確率で抽出する(古いサンプルも間引いて調整する)

**Reply by penginn-net** [reply]
フォロー中ユーザー(全員)の最新n件のリアクションを引っ張ってくるとかで良いのかしら？

**Reply by penginn-net** [reply]
手動追加なら重み付け変えるとかもいいかも

**Reply by penginn-net** [reply]
リノートしたかどうかも判断基準に加えたい

**Reply by penginn-net** [reply]
これ取りに行くだと辛そうRedisにつめつめする感じにしたい

**Reply by penginn-net** [reply]
リアクション非対応(likeのみ)のサーバーのユーザーへのフォロー多い人とか♥に埋め尽くされない？

**Reply by kozakura913** [reply]
ユーザー指定はホワイトリスト、ブラックリスト両方あった方が良さそう
> リアクション非対応(likeのみ)のサーバーのユーザーへのフォロー多い人とか♥に埋め尽くされない？

**Comment by penginn-net** [comment]
考えてたやつ
自分の投稿n件から頻出単語を抽出してそこから検索かける

**Reply by penginn-net** [reply]
好きな事を常に投稿してるならいいけど
あまり投稿してないとうーんかも

**Reply by kozakura913** [reply]
検索重いしFTTみたいに投稿飛んできた時に流し込む感じにしたい

**Reply by kozakura913** [reply]
単語一致だとアンテナでいいじゃんになる

**Comment by penginn-net** [comment]
自分の考えで行くとopensearch必要率が増える

**Comment by penginn-net** [comment]
自動更新なんてできなさそう

負荷高そう
遡る専用TL
Redisに少し長めにキャッシュ持ちたい

**Comment by kozakura913** [comment]
アンテナみたいな感じの条件付きTLをベースに条件をリノートされた投稿の添付ファイル詳細とかリアクション、自分が付けたリアクション、フォローユーザーがよく使うリアクションなど考慮で定期的に更新する感じ
よく使うハッシュタグも考慮？
頻出単語は投稿文から単語の抽出する処理が重そうだからやりたくない

**Comment by kozakura913** [comment]
GTL眺めてたら画像添付のみの検索性最悪な投稿が結構流れてて困る
